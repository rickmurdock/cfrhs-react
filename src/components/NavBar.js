import React from 'react';
import { NavLink } from 'react-router-dom';

const NavBar = () => {
  return (
    <nav className="navbar sticky-top navbar-toggleable-md">
      <button className="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarNavDropdown" aria-controls="navbarNavDropdown" aria-expanded="false" aria-label="Toggle navigation">
        <span><i className="fa fa-bars" aria-hidden="true"></i></span>
      </button>
      <span className="navbar-text">&nbsp;</span>
      <div className="collapse navbar-collapse" id="navbarNavDropdown">
        <div className="navbar-nav mr-auto">
          <NavLink activeClassName="selected" className="nav-item nav-link" exact to="/">Home</NavLink>
          <div className="nav-item dropdown">
            <a className="nav-link dropdown-toggle" href="" id="navbarDropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">About</a>
            <div className="dropdown-menu" aria-labelledby="navbarDropdownMenuLink">
              <NavLink activeClassName="selected" className="dropdown-item" to="/history">Our History</NavLink>
              <NavLink activeClassName="selected" className="dropdown-item" to="/board">Board of Directors</NavLink>
              <NavLink activeClassName="selected" className="dropdown-item" to="/membership">Society Membership</NavLink>
            </div>
          </div>
          <NavLink activeClassName="selected" className="nav-item nav-link" to="/museum">Museum</NavLink>
          <NavLink activeClassName="selected" className="nav-item nav-link" to="/gallery">Gallery</NavLink>
          <NavLink activeClassName="selected" className="nav-item nav-link" to="/shop">Gift Shop</NavLink>
          <NavLink activeClassName="selected" className="nav-item nav-link" to="/newsletter">Newsletter</NavLink>
          <NavLink activeClassName="selected" className="nav-item nav-link" to="/links">Links</NavLink>
          <NavLink activeClassName="selected" className="nav-item nav-link" to="/contact">Contact</NavLink>
        </div>
        <div className="navbar-brand my-auto">
          <a href="https://www.facebook.com/CentralFloridaRailroadMuseum/" target="_blank" rel="noopener noreferrer"><i className="fa fa-facebook" aria-hidden="false"></i></a>
        </div>
      </div>
    </nav>
  );
};

export default NavBar;