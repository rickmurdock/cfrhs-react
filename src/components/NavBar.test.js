import React from 'react';
import { MemoryRouter as Router, Link } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavBar from './NavBar';

test('NavBar renders correctly', () => {
    const tree = renderer.create(
        <Router><NavBar /></Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});