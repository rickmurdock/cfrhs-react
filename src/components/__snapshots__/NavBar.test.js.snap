// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<nav
  className="navbar sticky-top navbar-toggleable-md"
>
  <button
    aria-controls="navbarNavDropdown"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler navbar-toggler-right"
    data-target="#navbarNavDropdown"
    data-toggle="collapse"
    type="button"
  >
    <span>
      <i
        aria-hidden="true"
        className="fa fa-bars"
      />
    </span>
  </button>
  <span
    className="navbar-text"
  >
    Â 
  </span>
  <div
    className="collapse navbar-collapse"
    id="navbarNavDropdown"
  >
    <div
      className="navbar-nav mr-auto"
    >
      <a
        aria-current="true"
        className="nav-item nav-link selected"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
      <div
        className="nav-item dropdown"
      >
        <a
          aria-expanded="false"
          aria-haspopup="true"
          className="nav-link dropdown-toggle"
          data-toggle="dropdown"
          href=""
          id="navbarDropdownMenuLink"
        >
          About
        </a>
        <div
          aria-labelledby="navbarDropdownMenuLink"
          className="dropdown-menu"
        >
          <a
            aria-current={false}
            className="dropdown-item"
            href="/history"
            onClick={[Function]}
            style={undefined}
          >
            Our History
          </a>
          <a
            aria-current={false}
            className="dropdown-item"
            href="/board"
            onClick={[Function]}
            style={undefined}
          >
            Board of Directors
          </a>
          <a
            aria-current={false}
            className="dropdown-item"
            href="/membership"
            onClick={[Function]}
            style={undefined}
          >
            Society Membership
          </a>
        </div>
      </div>
      <a
        aria-current={false}
        className="nav-item nav-link"
        href="/museum"
        onClick={[Function]}
        style={undefined}
      >
        Museum
      </a>
      <a
        aria-current={false}
        className="nav-item nav-link"
        href="/gallery"
        onClick={[Function]}
        style={undefined}
      >
        Gallery
      </a>
      <a
        aria-current={false}
        className="nav-item nav-link"
        href="/shop"
        onClick={[Function]}
        style={undefined}
      >
        Gift Shop
      </a>
      <a
        aria-current={false}
        className="nav-item nav-link"
        href="/newsletter"
        onClick={[Function]}
        style={undefined}
      >
        Newsletter
      </a>
      <a
        aria-current={false}
        className="nav-item nav-link"
        href="/links"
        onClick={[Function]}
        style={undefined}
      >
        Links
      </a>
      <a
        aria-current={false}
        className="nav-item nav-link"
        href="/contact"
        onClick={[Function]}
        style={undefined}
      >
        Contact
      </a>
    </div>
    <div
      className="navbar-brand my-auto"
    >
      <a
        href="https://www.facebook.com/CentralFloridaRailroadMuseum/"
      >
        <i
          aria-hidden="false"
          className="fa fa-facebook"
        />
      </a>
    </div>
  </div>
</nav>
`;
